utils/mediaupload.js

```diff
@@ -3,6 +3,11 @@
  */
 import { compact, forEach, get, includes, noop, startsWith } from 'lodash';
 
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '../node_modules/@wordpress/i18n';
```

Comment:
Couldn't figure out why it's unable to resolve the module normally with just `from '@wordpress/i18n'`

Reply:
I'm not certain of the reason why  '@wordpress/i18n' can not be used here I don't think we have a circular dependency. It may be something intentional to keep utils generic but this util here is something specific to WordPress so it makes sense to import @wordpress/i18n here.
We already have '@wordpress/is-shallow-equal' so another option for us is to have '@wordpress/mediaupload'.
@gziolo would it be possible to share your thoughts on this? Are you aware of the reason why '@wordpress/i18n' cannot be imported in utils/mediaupload.js?

Reply:
Just to make sure, does this problem reproduce in your environments? @nb tells me a normal `import { __, sprintf } from '@wordpress/i18n';` here is working for him.

Reply:
Yes, I'm able to reproduce the problem. Everything seems to work until an error message is triggered.
To easily trigger error messages we can limit file uploads on the site to 3MB by adding the following line to .htaccess:
```
php_value upload_max_filesize 3M
php_value post_max_size 3M
```

Reply:
Maybe we are in the presence of an intermitent error or something that just happens in some setups.

Reply:
I did a quick check of all the things we’re setting as `@wordpress`-prefixed externals in the webpack config, to see if they can be imported correctly from `utils/index.js`. The majority of them cannot. Here are the results in case someone has a clue or encounters the same issue:

```
Can be imported:
- editor
- blob
- deprecated
- dom
- api-request
- a11y

Can’t be imported:
- blocks
- components
- viewport
- plugins
- edit-post
- core-blocks
- nux
- data
- date
- element
- core-data
- dom-ready
- hooks
- i18n
- is-shallow-equal
```

Does this need a bug report? This won't be a problem for this PR since we're moving mediaupload.js out of utils, but it could potentially create some hard-to-catch bugs down the line, since they don't trigger an error immediately on build.

Reply:
I don't there is anything to report. It all boils down to the fact that each module need to specify its dependencies when enqueuing the scripts. See: 
https://github.com/WordPress/gutenberg/blob/master/lib/client-assets.php#L188

```php
array( 'lodash', 'wp-blob', 'wp-deprecated', 'wp-dom', 'wp-api-request' ),
```

The list you shared more or less aligns with the dependencies set for this module.

In general, we are trying to split `utils` into smaller modules, so it will no longer existing a few weeks from now.